src/micromamba/install.sh

```diff
@@ -52,47 +61,54 @@ check_packages() {
     fi
 }
 
-check_git() {
-    if [ ! -x "$(command -v git)" ]; then
-        check_packages git
-    fi
-}
-
-find_version_from_git_tags() {
-    local variable_name=$1
-    local requested_version=${!variable_name}
-    if [ "${requested_version}" = "none" ]; then return; fi
-    local repository=$2
-    local prefix=${3:-"tags/v"}
-    local separator=${4:-"."}
-    local last_part_optional=${5:-"false"}
-    if [ "$(echo "${requested_version}" | grep -o "." | wc -l)" != "2" ]; then
-        local escaped_separator=${separator//./\\.}
-        local last_part
-        if [ "${last_part_optional}" = "true" ]; then
-            last_part="(${escaped_separator}[0-9]+)*?"
-        else
-            last_part="${escaped_separator}[0-9]+"
-        fi
-        local regex="${prefix}\\K[0-9]+${escaped_separator}[0-9]+${last_part}$"
-        local version_list
-        check_git
-        check_packages ca-certificates
-        version_list="$(git ls-remote --tags "${repository}" | grep -oP "${regex}" | tr -d ' ' | tr "${separator}" "." | sort -rV)"
-        if [ "${requested_version}" = "latest" ] || [ "${requested_version}" = "current" ] || [ "${requested_version}" = "lts" ]; then
-            declare -g "${variable_name}"="$(echo "${version_list}" | head -n 1)"
-        else
-            set +e
-            declare -g "${variable_name}"="$(echo "${version_list}" | grep -E -m 1 "^${requested_version//./\\.}([\\.\\s]|$)")"
-            set -e
-        fi
-    fi
-    if [ -z "${!variable_name}" ] || ! echo "${version_list}" | grep "^${!variable_name//./\\.}$" >/dev/null 2>&1; then
-        echo -e "Invalid ${variable_name} value: ${requested_version}\nValid values:\n${version_list}" >&2
-        exit 1
-    fi
-    echo "${variable_name}=${!variable_name}"
-}
+# # I don't think we should use Git. (See next comment.)
+# check_git() {
+#     if [ ! -x "$(command -v git)" ]; then
+#         check_packages git
+#     fi
+# }
+
+# # This approach is unfortunately seriously flawed. The reason is that micromamba
+# # is compiled by Conda-Forge and published on anaconda.org. The website
+# # <https://micro.mamba.pm> actually redirects to anaconda.org for the actual download.
+# # There is often several days of delay between when a new version of micromamba is
+# # published on GitHub and when it is available on anaconda.org. During this time,
+# # the download will be missing, and this procedure will fail.
```

Comment:
If so, we need to add a process to check the validity of the version.
I think it is not user friendly to use while specifying the complete version all the time without soft version matching existing.

Reply:
For now, it may be a good thing to remove soft version matching and basically use only latest as a basic strategy.

Reply:
Since https://micro.mamba.pm accepts `latest` as a version, why do we need to check its validity?

I'm in favor of using `latest` as default, with the ability to pin an exact version number if desired.

> For now, it may be a good thing to remove soft version matching

Ah, I see, thanks for pointing that out. It looks like the regex is allowing one to specify version X.Y to match the X.Y.Z release with the latest Z. So we would have to make clear in the docs that only exact versions are supported.

Reply:
I assume that we may want to specify as `"version": "1"` to avoid breaking changes.
However, since micromamba is just a package manager and the possibility of API changes is low, I don't see much of a problem if we can only specify `"latest"`.